migrations/4_stablecoin_converter.js

```diff
@@ -0,0 +1,23 @@
+/*eslint no-undef: "off"*/
+
+const BiMap = artifacts.require("IdToAddressBiMap.sol")
+const IterableAppendOnlySet = artifacts.require("IterableAppendOnlySet.sol")
+const StablecoinConverter = artifacts.require("StablecoinConverter.sol")
+const ERC20Mintable = artifacts.require("ERC20Mintable.sol")
+
+module.exports = async function (deployer, network) {
+  let fee_token
+  if (network == "development" || network == "coverage") {
+    await deployer.deploy(ERC20Mintable)
+    fee_token = await ERC20Mintable.deployed()
+  } else {
+    throw (`No migration found for network "${network}"`)
+  }
+
+  await deployer.deploy(BiMap)
+  await deployer.deploy(IterableAppendOnlySet)
+
+  await deployer.link(BiMap, StablecoinConverter)
+  await deployer.link(IterableAppendOnlySet, StablecoinConverter)
+  await deployer.deploy(StablecoinConverter, 2 ** 16 - 1, 1000, fee_token.address)
```

Comment:
should this be parametrized? or at least defined as a constant

Reply:
Sure, although as long as a value is only used once, I tend to not put into a constant since it adds another jump/context switch when trying to understand the code (now you have to scroll up to see what this value is).

Totally agree to use constants when the value is used more than once (I'd argue the value in the unit tests is different).

Reply:
I say this because I had to go to the smart contract, look for the constructor, and see the parameter. 

If those parameters had a constant it would read as:
```js
await deployer.deploy(StablecoinConverter, MAX_TOKENS, FEE_DENOMINATOR, fee_token.address)
```

Alternatively I find it ok in this case to do:
```js
const maxTokens = 2 ** 16 - 1
const feeDenominator = 1000 // fee is (1 / feeDenominator)
await deployer.deploy(StablecoinConverter, maxTokens, feeDenominator, fee_token.address)
```

Reply:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F3o7TKNLBKhbBNfPhEk%2Fgiphy.gif&token=fair" data-gitmeme-token="fair"><img src="https://media2.giphy.com/media/3o7TKNLBKhbBNfPhEk/giphy.gif" title="Created by gitme.me with /fair"/></a>